components/LearnPage/FillInTheBlankQuestion.tsx

```diff
@@ -0,0 +1,196 @@
+import React, { useState, useEffect } from "react";
+import { Box, Typography, TextField, Tooltip, Button } from "@mui/material";
+import styled from "@emotion/styled";
+import { blankAnswer } from "components/ContentEditor/components/exportStructures";
+import theme from "constants/theme";
+
+interface FillIntheBlankProps {
+    questionSegments: string[];
+    correctAnswers: blankAnswer[];
+    number: number;
+}
+
+const QuestionHolder = styled.div((props: any) => ({
+    width: "70%",
+    marginTop: 50,
+    height: "auto",
+    borderRadius: 4,
+    display: "flex",
+    flexDirection: "column",
+    boxShadow: "rgba(99, 99, 99, 0.2) 0px 0px 8px 0px",
+    transition: "box-shadow 0.3s ease-in-out, transform 0.2s ease-in-out",
+    alignSelf: "center",
+    ":hover": {
+        transform: props.answered && props.correct ? "" : "scale(1.01)",
+        boxShadow:
+            props.answered && props.correct
+                ? "rgba(99, 99, 99, 0.2) 0px 0px 8px 0px"
+                : "rgba(99, 99, 99, 0.1) 0px 0px 20px 0px",
+    },
+    padding: 20,
+}));
+
+const BlankAnswerTextField = styled(TextField)((props: any) => ({
+    "& .MuiFilledInput-root": {
+        backgroundColor: theme.palette.primary.light
+    },
+    "& .MuiFilledInput-root:hover": {
+        backgroundColor: theme.palette.primary.light
+    },
+    "& .MuiFilledInput-root.Mui-focused": {
+        backgroundColor: theme.palette.primary.light
+    },
+    marginBottom: props.answered ? 6 : 10,
+    width: 175
+}));
+
+const CorrectAnswerTextField = styled(TextField)({
+    '& fieldset': {
+        border: 'none',
+        display: 'inline'
+    },
+    '& .MuiInputBase-root.Mui-disabled': {
+        backgroundColor: '#22B16E',
+    },
+    "& .MuiInputBase-input.Mui-disabled": {
+        WebkitTextFillColor: "white",
+    },
+    marginBottom: 10,
+    width: 175
+});
+
+const QuestionSegments = styled(Typography)({
+    fontWeight: 200,
+    fontFamily: "DM Serif Display",
+    display: 'inline',
+    whiteSpace: 'pre-wrap'
+});
+
+function FillIntheBlankQuestion(props: FillIntheBlankProps) {
+    const [answerInputs, setAnswerInputs] = useState<string[]>([]);   // string array of question attempts/inputs
+    const [results, setResults] = useState<boolean[]>([]);   // boolean array of answer correctness
+    const [correct, setCorrect] = useState(false);    // Whether all answers are correct
+    const [answered, setAnswered] = useState(false);    // Whether the question was attempted yet
+
+    useEffect(() => {
```

Comment:
you can remove this useeffect and instead just initialize results to an array of passed in length of all false in the useState